PHYSICSNEMO_CODE_STRUCTURE.md

```diff
@@ -0,0 +1,692 @@
+
+---
+
+## Coding Standards
+
+**Location:** `CODING_STANDARDS/`
+
+- `MODELS_IMPLEMENTATION.md` - Model implementation guidelines
+- `EXTERNAL_IMPORTS.md` - External dependency guidelines
+
+---
+
+## Key Features
+
+### 1. Modular Architecture
+- Plug-and-play model components
+- Extensible datapipe system
+- Customizable metrics
+
+### 2. Distributed Training
+- Multi-GPU support
+- Domain decomposition
+- Efficient communication primitives
+
+### 3. Multiple Domains
+- Weather & climate forecasting
+- Computational fluid dynamics
+- Structural mechanics
+- Molecular dynamics
+- Geophysics
+- Healthcare
+
+### 4. State-of-the-Art Models
+- 20+ pre-implemented architectures
+- Latest research implementations
+- Production-ready models
+
+### 5. Comprehensive Testing
+- 281 test files
+- Pre-trained checkpoints
+- CI/CD integration
+
+### 6. Rich Examples
+- 776 example files
+- Domain-specific tutorials
+- Production use cases
+
+---
+
+## Dependencies
+
+### Core Dependencies (from `pyproject.toml`)
+- **PyTorch** ≥ 2.4.0 - Deep learning framework
+- **Warp-lang** - NVIDIA's high-performance Python framework
+- **Hydra-core** ≥ 1.3.2 - Configuration management
+- **OmegaConf** ≥ 2.3.0 - Configuration system
+- **xarray** ≥ 2025.6.1 - N-D labeled arrays
+- **zarr** ≥ 2.18.3 - Chunked array storage
+- **h5py** ≥ 3.15.1 - HDF5 file I/O
+- **einops** ≥ 0.8.1 - Tensor operations
+- **timm** ≥ 1.0.22 - Vision models
+- **onnx** ≥ 1.14.0 - Model export
+- **pandas** - Data analysis
+- **s3fs** ≥ 2023.5.0 - S3 filesystem
+- **tqdm** ≥ 4.60.0 - Progress bars
+- **requests** ≥ 2.32.2 - HTTP library
+
+---
+
+## Version Information
+
+- **Current Version:** 1.4.0a0 (alpha)
+- **Branch:** v2.0-refactor
+- **Python:** 3.11+
+- **License:** Apache 2.0 (see LICENSE.txt)
+
+---
+
+## Getting Started
+
+### Installation
+```bash
+# Clone the repository
+git clone -b v2.0-refactor https://github.com/NVIDIA/physicsnemo.git
+cd physicsnemo
+
+# Install in development mode
+pip install -e .
+```
+
+### Quick Example
+```python
+import physicsnemo
+from physicsnemo.models import FNO
+
+# Create a Fourier Neural Operator
+model = FNO(
+    in_channels=2,
+    out_channels=1,
+    decoder_layer_size=128,
+    num_fno_layers=4,
+    num_fno_modes=12,
+)
+
+# Model is ready for training or inference
+print(f"PhysicsNeMo version: {physicsnemo.__version__}")
+```
+
+---
+
+## Resources
+
+- **Documentation:** `docs/`
+- **Examples:** `examples/`
+- **Tests:** `test/`
+- **FAQ:** `FAQ.md`
+- **Contributing:** `CONTRIBUTING.md`
+- **Migration Guide:** `v2.0-MIGRATION-GUIDE.md`
+
+---
+
+## File Statistics
+
+- **Total Python files:** 298 in `physicsnemo/`, 378 in `examples/`, 191 in `test/`
+- **Total example files:** 776 (includes YAML configs, scripts, docs)
+- **Total test files:** 281 (includes checkpoints and test data)
+- **Documentation files:** 196 (RST, Python, images)
+- **Lines of code:** ~100,000+ (estimated)
```

Comment:
These will go out-of-sync quickly. Same for other parts of the document (e.g., pyproject.toml contents, version pins, etc.) Are we sure we want to have this written down here?